src/pretrain_utils.py

```diff
@@ -54,37 +63,101 @@ def input_fn_builder(tokenizer: tf_text.BertTokenizer,
 
   mask_values_chooser = tf_text.MaskValuesChooser(len(vocab),
                                                   mask_token_id, 0.8)
-  num_patch_per_row = input_config.image_size // input_config.patch_size
+  num_patch_per_row = image_size // patch_size
+
+  def make_masked_patch_prediction_target(patch_tokens):
+    """Make taget labels for masked patch prediction
+
+    Args:
+      patch_tokens: <tf.Tensor>[batch, sequence_length, patch_token_size]
+
+    Returns:
+      <tf.Tensor>[batch, sequence_length]
+    """
 
-  def make_masked_lm_features(features: Mapping[str, tf.Tensor]):
-    """Randomly mask out tokens
+    channels = get_shape_list(patch_tokens)[-1] // (patch_size**2)
+    bin_size = max_pixel_val // (2 ** output_channel_bits)
```

Comment:
Maybe we could go over the code a bit together to help me understand the details.